icu4c/source/test/fuzzer/dtfmtsym_fuzzer.cpp

```diff
@@ -0,0 +1,75 @@
+// © 2023 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// Fuzzer for ICU DateFormatSymbols.
+
+#include <cstring>
+
+#include "fuzzer_utils.h"
+
+#include "unicode/dtfmtsym.h"
+#include "unicode/locid.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+    uint16_t rnd;
+    icu::DateFormatSymbols::DtContextType context;
+    icu::DateFormatSymbols::DtWidthType width;
+    if (size < sizeof(rnd) + sizeof(context) + sizeof(width)) return 0;
+    icu::StringPiece fuzzData(reinterpret_cast<const char *>(data), size);
+
+    std::memcpy(&rnd, fuzzData.data(), sizeof(rnd));
+    icu::Locale locale = GetRandomLocale(rnd);
+    fuzzData.remove_prefix(sizeof(rnd));
+
+    std::memcpy(&context, fuzzData.data(), sizeof(context));
+    fuzzData.remove_prefix(sizeof(context));
+    icu::DateFormatSymbols::DtContextType context_mod =
+        static_cast<icu::DateFormatSymbols::DtContextType>(
+            context % icu::DateFormatSymbols::DtContextType::DT_CONTEXT_COUNT);
```

Comment:
With both the context and the width, do you want to do the "mod" thing (or _only_ do the "mod" thing)?  It seems again that it'd be useful also to check that passing a completely illegal enum value (with, of course, you can do in C++ with casting) doesn't cause anything to barf.